templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,453 @@
+            ))),
+            InvokeContractResult::Success {
+                return_value: _,
+                events: _,
+                used_energy,
+            } => {
+                let e = used_energy.energy;
+                println!("Contract invoke success: estimated_energy={e}");
+                Ok(Energy {
+                    energy: e,
+                })
+            }
+        }
+    }
+
+    /// A function to get the current nonce of the wallet account.
+    pub async fn get_nonce(
+        &mut self,
+        address: AccountAddress,
+    ) -> Result<AccountNonceResponse, DeployError> {
+        let nonce = self.client.get_next_account_sequence_number(&address).await?;
+        Ok(nonce)
+    }
+
+    /// A function that checks the outcome of the deploy transaction.
+    /// It throws if the `block_item` is not a deploy transaction.
+    /// It returns the error code if the transaction reverted.
+    fn check_outcome_of_deploy_transaction(
+        &self,
+        block_item: BlockItemSummary,
+    ) -> Result<ModuleDeployed, DeployError> {
+        match block_item.details {
+            BlockItemSummaryDetails::AccountTransaction(a) => match a.effects {
+                AccountTransactionEffects::None {
+                    transaction_type,
+                    reject_reason,
+                } => {
+                    if transaction_type != Some(TransactionType::DeployModule) {
+                        return Err(DeployError::InvalidBlockItem(
+                            "Expected transaction type to be DeployModule".into(),
+                        ));
+                    }
+
+                    Err(DeployError::TransactionRejectedR(format!(
+                        "Module deploy rejected with reason: {reject_reason:?}"
+                    )))
+                }
+                AccountTransactionEffects::ModuleDeployed {
+                    module_ref,
+                } => Ok(ModuleDeployed {
+                    module_ref,
+                }),
+                _ => Err(DeployError::InvalidBlockItem(
+                    "The parsed account transaction effect should be of type `ModuleDeployed` or \
+                     `None` (in case the transaction reverted)"
+                        .into(),
+                )),
+            },
+            _ => Err(DeployError::InvalidBlockItem(
+                "Can only parse an account transaction (no account creation transaction or chain \
+                 update transaction)"
+                    .into(),
+            )),
+        }
+    }
+
+    /// A function that checks the outcome of the initialization transaction.
+    /// It throws if the `block_item` is not a initialization transaction.
+    /// It returns the error code if the transaction reverted.
+    fn check_outcome_of_initialization_transaction(
+        &self,
+        block_item: BlockItemSummary,
+    ) -> Result<ContractInitialized, DeployError> {
+        match block_item.details {
```

Comment:
hmm, `is_rejected_account_transaction()` just returns a boolean if the transaction got rejected. The developer writing such scripts most likely wants to know the `reject_reason` to be able to debug their deployment script. 
To a smaller extent, if we use the `contract_init()` function, we also could just return some minimal information about that the given tx is not a `contract init transaction` in case the function returns `None`. 

The idea with these verbose errors is to give the developer as much as possible feedback on what could have been wrong in the transaction in case it failed.

Reply:
https://docs.rs/concordium-rust-sdk/latest/concordium_rust_sdk/types/struct.BlockItemSummary.html#method.is_rejected_account_transaction

This function returns the reject reason which I think is what you want to print if I understood correctly.

Reply:
ok, I see we can simplify the `check_outcome_of_init_transaction`. To do the same for the `check_outcome_of_update_transaction`/`check_outcome_of_deploy_transaction` function we would need something like `contract_update()`/`contract_deploy()`. So to keep the three parsing functions consistent, I would think it makes sense to keep the old logic. What do you think?